connector/oidc/oidc.go

```diff
@@ -352,6 +394,20 @@ func (c *oidcConnector) LoginURL(s connector.Scopes, callbackURL, state string)
 	if s.OfflineAccess {
 		opts = append(opts, oauth2.AccessTypeOffline, oauth2.SetAuthURLParam("prompt", c.promptType))
 	}
+
+	if c.pkceChallenge != "" {
+		switch c.pkceChallenge {
+		case codeChallengeMethodPlain:
+			c.pkceVerifier = oauth2.GenerateVerifier()
```

Comment:
Oh no, verifier is not the provider property. It seems when two users login at the same time, it will not work

see https://github.com/dexidp/dex/pull/3188#discussion_r1443421874